frontend/components/idioms.jsx

```diff
@@ -145,216 +126,309 @@ export function UntraceableState() {
       <span>Loading: {loading}</span>
       Result:{result}
     </div>
-  )
+  );
 }
 
 export function CrudeDeclarations() {
-  const calendarDays = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30];
-  return (<ol>
-    {calendarDays.map((val) => <span key={val}>{val}</span>)}
-  </ol>)
+  const calendarDays = [...Array(30).keys()]; // Less prone to error, cleaner
+  return (
+    <ol>
+      {calendarDays.map((val) => (
+        <span key={val}>{val}</span>
+      ))}
+    </ol>
+  );
 }
 
 export function MagicNumbers(age) {
-  return (<ol>
-    { age < 18 ? <div>Spicy</div> : <div>You are not old enough</div>}
-  </ol>)
+  // Set a varible instead of hardcoded number;
+  const minor = age < 18;
+  return (
+    <ol>{minor ? <div>Spicy</div> : <div>You are not old enough</div>}</ol>
+  );
 }
 
 export function UnidiomaticHTMLStructure() {
-  const [name, setName] = useState("")
-  const handleSubmit = (e) => {}
-
-  return (<div>
-    <input value={name} name="name" type="text" onChange={setName} />
-    <button type="submit" onClick={handleSubmit}>Submit</button>
-  </div>)
+  const [name, setName] = useState('');
+  const handleSubmit = (e) => {};
+  //Fix to a form with a proper onchange
```

Comment:
Why is it proper? (it's also not an onchange)

Reply:
Fixed the handle submit to be part of a form instead of a div, in which the button has a type of submit and the form uses onSubmit fixing the HTML structure